### ISSUE TITLE
Setting values with `set`.

### ISSUE THREAD

**Issue by isaacazuelos** [isaacazuelos/keep#2]
`keep set set.some.keypath 123`

There are a lot of difficult issues here:

- Data structure traversal and mutation can be tricky to get right without blowing the stack.
- What's the difference between _setting_ a value, and _merging_ in a value.
- How do we handle non-existent keys?
- How do we know how to parse the new values?

**Comment by isaacazuelos** [comment]
Suppose we do the following:

```sh
keep new test
keep set test a.b 2
keep set test a "{\"a\": 1}"
```

- Should the output of `keep get test a` be `{a: 1}` or `{a: 1, b:2}`? 
- How does this interact with `--preserve`?
- What does this mean if the value being set is a list? 
- Do we need an `unset` command?

**Comment by isaacazuelos** [comment]
Closed by 4e7b5106b6a35976111cebb1142941ef4b209b0f.
